cmd/forward/main.go

```diff
@@ -246,6 +273,10 @@ func forwardQuery(ctx context.Context, queryState *presto.QueryStateInfo, client
 				failed.Add(1)
 				return
 			}
+			//build cache for running query
+			if clientResult.NextUri != nil {
+				*cachedQueries = append(*cachedQueries, QueryCache{QueryId: clientResult.Id, NextUri: *clientResult.NextUri, Client: client})
```

Comment:
Here you write to cachedQueries concurrently, which could lead to issues.
In the comment above I requested to pre-allocate the elements. so here you can just update element using index `i-1`, without needing to update the cachedQueries value